Makefile

```diff
@@ -73,9 +74,14 @@ install-zlib:
 check: check-ltag check-dco check-lint check-flatc
 
 flatc:
-	flatc -o $(CURDIR)/soci/fbs -g $(CURDIR)/soci/fbs/ztoc.fbs
-
-check-flatc: flatc ## check if protobufs needs to be generated again
+	rm -r $(CURDIR)/soci/fbs/ztoc
+	flatc -o $(CURDIR)/soci/fbs -g $(FBS_FILE_PATH)
+
+# check if flatbuffers needs to be generated again
+check-flatc:
+	$(eval TMPDIR := $(shell mktemp -d))
+	flatc -o $(TMPDIR) -g $(FBS_FILE_PATH)
+	diff -qr $(TMPDIR)/ztoc $(CURDIR)/soci/fbs/ztoc || (printf "\n\nThe Ztoc schema seems to be modified. Please run 'make flatc' to re-generate Go files\n\n"; exit 1)
 
```

Comment:
Nit: If you add an `rm -rf $(TMPDIR)` here, then we don't have to worry about this filling up /tmp. (Yes, the cron job is supposed to clean it up eventually...as long as the system actually has a running tmpfile clean script running and properly configured. That should generally be treated as a failsafe, not something you can count on.)

It's okay to leave the directory behind in the case where the diff fails, since the user may want to inspect it.

Reply:
Done.